internal/cli/cli.go

```diff
@@ -23,11 +28,31 @@ func NewRootCommand() *cobra.Command {
 
 	cmd.PersistentFlags().String("log-level", "info", "Configure log level")
 	cmd.PersistentFlags().String("listen-address", ":8080", "Server listen address")
+	cmd.PersistentFlags().Bool("tls-skip-verify", false, "Skip TLS certificate verification")
+	cmd.PersistentFlags().String("grafana-proxy-url", "", "Grafana url to proxy to")
+	cmd.PersistentFlags().String("grafana-client-url", "", "Grafana url to connect with client")
+	cmd.PersistentFlags().String("cookie-name", "", "Cookie name with jwt token. If set will take precedence over auth header")
+	cmd.PersistentFlags().String("admin-user", "admin", "Admin user")
+	cmd.PersistentFlags().String("admin-password", "", "Admin password")
 
 	return cmd
 }
 
 func (c *RootCommand) persistentPreRunE(cmd *cobra.Command, args []string) error {
+
+	// Read from config
+	viper.SetConfigName("config")
+	viper.AddConfigPath(".")
+	viper.AddConfigPath("/etc/grafana-auth-proxy/")
+	viper.WatchConfig()
```

Comment:
I haven't fully tested this but the feature I'm looking to have is us modifying the configmap to have a new group mapping, the file gets updated in the pod filesystem and viper re-reads the config without needing to restart the proxy. I can add a function hook to show a message in logs as is described [here](https://github.com/spf13/viper#watching-and-re-reading-config-files).

The groups config is only read when the proxy is activated, basically when a request comes, so config should be as updated as possible to avoid a case where the user waits because a config hasn't been applied.

Reply:
To accomplish that you will need to look at using a sync.Map and how group and config information gets passed to the server also may need adjustment to prevent race conditions.